src/common/services/api/cart.service.js

```diff
@@ -82,9 +87,36 @@ class Cart {
       });
   }
 
-  addItem(uri, data){
+  getTotalQuantity() {
+    return this.cortexApiService
+      .get({path: ['carts', this.cortexApiService.scope, 'default']})
+      .map((cart) => {
+        return cart['total-quantity'];
+      });
+  }
+
+  addItem(uri, data) {
     data.quantity = 1;
 
+    if(this.sessionService.getRole() == Roles.public) {
+      return this.getTotalQuantity().mergeMap((total) => {
+        if(total <= 0) {
+          this.$cookies.remove(Sessions.cortex, {path: '/', domain: '.cru.org'});
+          this.$cookies.remove(Sessions.give, {path: '/', domain: '.cru.org'});
+          // Defer til next digest so $cookie.remove propagates.
+          return Observable.from(this.$timeout(angular.noop, 10)).mergeMap(() => {
+            return this.cortexApiService.post({
```

Comment:
I could DRY it up by moving it to another function. I tried assigning the Observable to a variable under the assumption that observables include setup/teardown, but since we are wrapping a promise, the request happens immediately when the observable is created, instead of when it is subscribed too.